core/src/test/java/hudson/util/io/TarArchiverTest.java

```diff
@@ -129,27 +127,28 @@ private static void run(FilePath dir, String... cmds) throws InterruptedExceptio
         }
     }
 
-    @Ignore("TODO fails to add empty directories to archive")
     @Issue("JENKINS-73837")
     @Test
     public void emptyDirectory() throws Exception {
         Path tar = tmp.newFile("test.tar").toPath();
         Path root = tmp.newFolder().toPath();
-        Files.createDirectory(root.resolve("foo"));
-        Files.createDirectory(root.resolve("bar"));
-        Files.writeString(root.resolve("bar/file.txt"), "foobar", StandardCharsets.UTF_8);
-        try (OutputStream out = Files.newOutputStream(tar)) {
-            new FilePath(root.toFile()).tar(out, "**");
+        Files.createDirectory(root.resolve("a"));
+        Files.createDirectory(root.resolve("b"));
+
+        Files.writeString(root.resolve("a/file.txt"), "empty_dir_test");
+
+        try (OutputStream outputStream = Files.newOutputStream(tar)) {
+            new FilePath(root.toFile()).tar(outputStream, "**");
         }
+
         Set<String> names = new HashSet<>();
-        try (InputStream is = Files.newInputStream(tar);
-                TarInputStream tis = new TarInputStream(is, StandardCharsets.UTF_8.name())) {
-            TarEntry te;
-            while ((te = tis.getNextEntry()) != null) {
-                names.add(te.getName());
+        try (TarInputStream tarInputStream = new TarInputStream(Files.newInputStream(tar), StandardCharsets.UTF_8.name())) {
```

Comment:
this input stream needs closing as well by assigning `Files.newInputStream(tar)` to a variable in the try with resources

Reply:
Fixed in latest commit.